packages/gatsby/src/utils/babel-config.js

```diff
@@ -155,6 +155,11 @@ module.exports = async function babelConfig(program, stage) {
     babelrc.plugins.unshift(`react-hot-loader/babel`)
   }
 
+  if (stage !== `develop`) {
+    // Remove prop-types in production.
+    babelrc.plugins.unshift(`transform-react-remove-prop-types`)
```

Comment:
Does this affect files parsed in source plugins? React-bootstrap uses/depends on some runtime propType logic and that would break the doc site i'm building with react-docgen :P